Orange/widgets/data/owgroupby.py

```diff
@@ -0,0 +1,490 @@
+
+    class Outputs:
+        data = Output("Data", Table, doc="Aggregated data")
+
+    class Error(OWWidget.Error):
+        unexpected_error = Msg("{}")
+
+    settingsHandler = DomainContextHandler()
+
+    gb_attrs: List[Variable] = ContextSetting([])
+    aggregations: Dict[Variable, Set[str]] = ContextSetting({})
+    auto_commit: bool = Setting(True)
+
+    def __init__(self):
+        super().__init__()
+        ConcurrentWidgetMixin.__init__(self)
+
+        self.data = None
+        self.result = None
+
+        self.gb_attrs_model = DomainModel(
+            valid_types=(DiscreteVariable, ContinuousVariable, StringVariable),
+            separators=False,
+        )
+        self.agg_table_model = VarTableModel(self)
+        self.agg_checkboxes = {}
+
+        self.__init_control_area()
+        self.__init_main_area()
+
+    def __init_control_area(self) -> None:
+        """Init all controls in the control area"""
+        box = gui.vBox(self.controlArea, "Group by")
+        self.gb_attrs_view = AggregateListViewSearch(
+            selectionMode=QListView.ExtendedSelection
+        )
+        self.gb_attrs_view.setModel(self.gb_attrs_model)
+        self.gb_attrs_view.selectionModel().selectionChanged.connect(self.__gb_changed)
+        box.layout().addWidget(self.gb_attrs_view)
+
+        gui.auto_send(self.buttonsArea, self, "auto_commit")
+
+    def __init_main_area(self) -> None:
+        """Init all controls in the main area"""
+        # aggregation table
+        self.agg_table_view = tableview = QTableView()
+        tableview.setModel(self.agg_table_model)
+        tableview.setSelectionBehavior(QAbstractItemView.SelectRows)
+        tableview.selectionModel().selectionChanged.connect(self.__rows_selected)
+        tableview.horizontalHeader().setSectionResizeMode(QHeaderView.Stretch)
+
+        vbox = gui.vBox(self.mainArea, " ")
+        vbox.layout().addWidget(tableview)
+
+        # aggregations checkboxes
+        grid_layout = QGridLayout()
+        gui.widgetBox(self.mainArea, orientation=grid_layout, box="Aggregations")
+        itm_col = ceil(len(AGGREGATIONS) / 3)
+
+        for i, agg in enumerate(AGGREGATIONS):
+            self.agg_checkboxes[agg] = cb = CheckBox(agg, self)
+            cb.setDisabled(True)
+            cb.stateChanged.connect(partial(self.__aggregation_changed, agg))
+            grid_layout.addWidget(cb, i % itm_col, i // itm_col)
+
+    ############
+    # Callbacks
+
+    def __rows_selected(self) -> None:
+        """Callback for table selection change; update checkboxes"""
+        selected_attrs = self.get_selected_attributes()
+
+        types_ = set(type(attr) for attr in selected_attrs)
+        active_aggregations = [self.aggregations[attr] for attr in selected_attrs]
+        for agg, cb in self.agg_checkboxes.items():
+            cb.setDisabled(len(types_ & AGGREGATIONS[agg].types) == 0)
+
+            activated = [agg in a for a in active_aggregations]
```

Comment:
Just an observation: it would be fun to use a set here, `activated = {agg in a for a in active_aggregations}`, because the condition would then be `Qt.checked if activated == {True} else (Qt.Unchecked if activated == {False} else Qt.PartiallyChecked)`. :)